wasm_src/carta_computation/post.ts

```diff
@@ -103,4 +107,48 @@ Module.GenerateVertexData = (sourceVertices: Float32Array, indexOffsets: Int32Ar
     return destHeapFloat;
 };
 
+Module.CalculateCatalogSize = (data: Float32Array, min: number, max: number, sizeMin: number, sizeMax: number, scaling: number, area: boolean, devicePixelRatio: number, alpha: number = 1000, gamma: number = 1.5): Float32Array => {
+    const N = data.length;
+    const bytes_per_element = data.BYTES_PER_ELEMENT;
+    const dataOnWasmHeap = Module._malloc(N * bytes_per_element);
+
+    Module.HEAPF32.set(data, dataOnWasmHeap / bytes_per_element);
+
+    if (area) {
+        calculateCatalogSizeArea(dataOnWasmHeap, N, min, max, sizeMin, sizeMax, scaling, devicePixelRatio, alpha, gamma);
+    } else {
+        calculateCatalogSizeDiameter(dataOnWasmHeap, N, min, max, sizeMin, sizeMax, scaling, devicePixelRatio, alpha, gamma);
+    }
+
+    const float32 = new Float32Array(Module.HEAPF32.buffer, dataOnWasmHeap, N);
+    Module._free(dataOnWasmHeap);
```

Comment:
technically, if we call `Module._free` on that pointer, then the `float32` array is now pointing to memory that can be freely used/overwritten

Reply:
this is ok for now, but please keep this in mind down the line. If we intersperse any calls to `CartaComputation` and our use of these results, undefined behaviour would ensue. The safe way to do this is to `slice()` the array, in order to move it from WASM back to javascript

Reply:
ah, Thanks for point it out.